### PR TITLE
[Silabs] [EFR32] Refactor 917 NCP and 917 SoC Bluetooth to use the same implementation

### PR DESCRIPTION
**Pull request by shgutte**
Problem
There are only some code changes in RS9116 and RS9117 BLE implementation, which is the code's repetition.

Fixes added
Refactored In such a way that one file will work for both RS9116 and RS9117 implementation. 

Testing
I tested with all combinations of RS9116 and RS9117.

### CODE REVIEW

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -27,8 +27,11 @@
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
 
 #include <platform/internal/BLEManager.h>
+#define RSI_BLE_ENABLE 1
```
Comment (mkardous-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:30]: This define should be removed. It is already being defined in the `wfx_sl_blie_init.h` header and in the sdk `.gni` files.

File: third_party/silabs/efr32_sdk.gni [hunk 2]
```diff
@@ -193,6 +193,7 @@ template("efr32_sdk") {
 
     if (chip_enable_ble_rs911x) {
       _include_dirs += [ "${wifi_sapi_root}/sapi/include" ]
+      _include_dirs += [ "${chip_root}/src/platform/silabs/rs911x" ]
```
Comment (mkardous-silabs) [inline third_party/silabs/efr32_sdk.gni:196]: why is this necessary? the files are already included in the `src/platform/silabs/.../BUILD.gn` files.

File: third_party/silabs/SiWx917_sdk.gni [hunk 3]
```diff
@@ -69,6 +69,7 @@ template("siwx917_sdk") {
       "${wisemcu_sdk_root}/third_party/freertos/include",
       "${wisemcu_sdk_root}/third_party/freertos/portable/GCC/ARM_CM4F",
       "${examples_plat_dir}/device/inc",
+      "${chip_root}/src/platform/silabs/rs911x",
```
Comment (mkardous-silabs) [inline third_party/silabs/SiWx917_sdk.gni:72]: same comment as `efr32_sdk.gni`

File: src/platform/silabs/rs911x/wfx_sl_ble_init.c [hunk 4]
```diff
@@ -301,9 +307,7 @@ void rsi_ble_add_char_val_att(void * serv_handler, uint16_t handle, uuid_t att_t
     new_att.property = val_prop;
 
     if (data != NULL)
-    {
         memcpy(new_att.data, data, RSI_MIN(sizeof(new_att.data), data_len));
-    }
```
Comment (mkardous-silabs) [inline src/platform/silabs/rs911x/wfx_sl_ble_init.c:306]: small nit, leave the `{}` coding standard wise

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 5]
```diff
@@ -114,11 +117,13 @@ void sl_ble_event_handling_task(void)
...
-        //! This semaphore is waiting for next ble event task
-        rsi_semaphore_wait(&sl_ble_event_sem, 0);
-
         // checking for events list
         event_id = rsi_ble_app_get_event();
+        if (event_id == -1)
+        {
+            rsi_semaphore_wait(&sl_ble_event_sem, 0);
+            continue;
+        }
```
Comment (mkardous-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:126]: I think this change makes the code less clear. I would use the initial implementation. if the event id is `-1` or `unknown`, it is managed by the switch-case default.

**Review by mkardous-silabs — CHANGES_REQUESTED** [review]

File: third_party/silabs/efr32_sdk.gni (hunk 2 above)
Comment (jmartinez-silabs) [reply third_party/silabs/efr32_sdk.gni:196]: I believe it is necessary because they have some definitions/ functions declared in `rsi_ble_config.h` and `wfx_sl_ble_init.h` that need to be known by wiseMCU/wiseConnect

There might be a cleaner way to do it tho. would need to investigate

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 5 above)
Comment (jmartinez-silabs) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp:126]: Why would we wait only after an event_id -1? Does `rsi_ble_app_get_event` block the task on its own? If it doesn't this task will continuously be active until a event_id -1.
 If `rsi_ble_app_get_event` does block. Why do we need a semaphore wait after a -1 event

Also what is a event_id -1, it's an error? an unhandled event? It is missing either a relevant name or a clear comment describing what you are doing here

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 1 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp:30]: Yeah fixed

File: third_party/silabs/efr32_sdk.gni (hunk 2 above)
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni:196]: Yes, @jmartinez-silabs is correct. and we will move all these files to third-party silabs after refactoring we already discussed this.

File: src/platform/silabs/rs911x/wfx_sl_ble_init.c (hunk 4 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/wfx_sl_ble_init.c:306]: updated

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 5 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp:126]: when we did not receive any event then this AP (rsi_ble_app_get_event ) will return -1 then we will wait till we get the event after getting the event we will continue. as @mkardous-silabs said and I'm also thinking the same initial implementation is better than this so updated the file

**Review by mkardous-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]
